app/controllers/api/base_controller.rb

```diff
@@ -1,3 +1,14 @@
 class Api::BaseController < ApplicationController
   layout nil
+
+  # Method to log login attempts
+  def log_login_attempt(user_id:, attempted_at:, successful:, ip_address:)
```

Comment:
Consider using a background job for logging login attempts if the volume is expected to be high. This would prevent the logging process from blocking the main thread and potentially affecting performance.